src/Interpreters/getTableExpressions.cpp

```diff
@@ -93,7 +94,7 @@ static NamesAndTypesList getColumnsFromTableExpression(
     {
         const auto table_function = table_expression.table_function;
         auto query_context = context->getQueryContext();
-        const auto & function_storage = query_context->executeTableFunction(table_function);
+        const auto & function_storage = query_context->executeTableFunction(table_function, select_query_hint);
```

Comment:
For anyone wondering about this change: The reason `select_query_hint` had to be passed in here, is to avoid a very narrow bug, that happens because `executeTableFunction(..)` caches it's result, but gives different results depending on whether `select_query_hint` is passed to it.

`InterpreterSelectQuery` would previously call `joined_tables.getLeftTableStorage()` and then `joined_tables.resolveTables()`.
It now calls `joined_tables.resolveTables()` and then `joined_tables.getLeftTableStorage()`.

This meant that previously it would first call:
`executeTableFunction(table_function, select_query_hint)`
and then:
`executeTableFunction(table_function) //returned cached result from previously.`

Because of the change in `InterpreterSelectQuery`, this order was swapped, which meant that the wrong result was cached. This affected a single test (02483_check_virtuals..), which did this query:

```sql
insert into test select *, _file, _path from file(0483_data.LineAsString);
```

Here schema inference would fail, because the single parameter `executeTableFunction(table_function)` call was cached.